crates/core/tedge_api/src/messages.rs

```diff
@@ -534,6 +535,83 @@ impl RestartOperationResponse {
     }
 }
 
+#[derive(Debug, Deserialize, Serialize, PartialEq, Copy, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
+pub enum CommandStatus {
+    Init,
+    Executing,
+    Successful,
+    Failed,
+}
+
+#[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct LogMetadata {
+    pub types: Vec<String>,
+}
+
+impl<'a> Jsonify<'a> for LogMetadata {}
+
+#[derive(Debug, Deserialize, Serialize, Eq, PartialEq, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct LogUploadCmdPayload {
+    pub status: CommandStatus, //Define a different enum if this op needs more states,
+    pub tedge_url: String,
+    #[serde(rename = "type")]
+    pub log_type: String,
+    #[serde(with = "time::serde::rfc3339")]
+    pub date_from: OffsetDateTime,
+    #[serde(with = "time::serde::rfc3339")]
+    pub date_to: OffsetDateTime,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub search_text: Option<String>,
+    pub lines: usize,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub reason: Option<String>,
+}
+
+impl<'a> Jsonify<'a> for LogUploadCmdPayload {}
+
+impl LogUploadCmdPayload {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        status: CommandStatus,
+        tedge_url: String,
+        log_type: String,
+        date_from: OffsetDateTime,
+        date_to: OffsetDateTime,
+        search_text: Option<String>,
+        lines: usize,
+        reason: Option<String>,
+    ) -> Self {
+        Self {
+            status,
+            tedge_url,
+            log_type,
+            date_from,
+            date_to,
+            search_text,
+            lines,
+            reason,
+        }
+    }
```

Comment:
IMHO this `new` method is trivial and completely unnecessary and it'd be better to use normal struct creation syntax because when having 8 arguments, it's better to spell each of them by name, and if you forget one, it tells you the missing field by name, whereas using a function the error is about `wrong number of arguments`, and you can also spell the fields in any order.

Reply:
That's a good point. Indeed, I mess-up `tedge_url` and `log_type` while rebasing this PR on top of the `tedge-log-plugin` PR where the proposal were to get these properties in a different order.

=> replaced by normal struct creation: https://github.com/thin-edge/thin-edge.io/pull/2108/commits/f4b45e6a9e2ea98927c55d80199e1aaeebeef322